server/api/src/rsDataObjPut.cpp

```diff
@@ -576,6 +577,8 @@ namespace
 
         try {
             dataObjInfo_t* dataObjInfoHead{};
+            irods::at_scope_exit free_data_object_info{[&dataObjInfoHead] { freeAllDataObjInfo(dataObjInfoHead); }};
```

Comment:
It looks like we might be able to remove `dataObjInfoHead` since it is not being used. `file_object_factory` frees the `DataObjInfo` internally if `nullptr` is passed as the 4th parameter (`nullptr` is also the default value). Up to you - this change has the same effect.

Reply:
Will take a look.

Reply:
I think I added this because a memory leak was detected when putting a directory which involved overwriting existing data objects.

I'm going to leave it as everything is happy at this time. We can continue to iterate on this after it is merged.